src/main/java/io/swagger/codegen/languages/DefaultCodegenConfig.java

```diff
@@ -2238,19 +2238,17 @@ public CodegenParameter fromParameter(Parameter parameter, Set<String> imports)
 
             codegenParameter.dataType = codegenProperty.datatype;
             codegenParameter.dataFormat = codegenProperty.dataFormat;
-            boolean isEnum = getBooleanValue(codegenProperty, IS_ENUM_EXT_NAME);
-            if(isEnum) {
+
+            if (getBooleanValue(codegenProperty, IS_ENUM_EXT_NAME)) {
                 codegenParameter.datatypeWithEnum = codegenProperty.datatypeWithEnum;
                 codegenParameter.enumName = codegenProperty.enumName;
-            }
-
-            // enum
-            updateCodegenPropertyEnum(codegenProperty);
-            codegenParameter.getVendorExtensions().put(CodegenConstants.IS_ENUM_EXT_NAME, Boolean.TRUE);
-            codegenParameter.getVendorExtensions().put(CodegenConstants.IS_LIST_CONTAINER_EXT_NAME, Boolean.TRUE);
-            codegenParameter._enum = codegenProperty._enum;
-            codegenParameter.allowableValues = codegenProperty.allowableValues;
 
+                updateCodegenPropertyEnum(codegenProperty);
+                codegenParameter.getVendorExtensions().put(CodegenConstants.IS_ENUM_EXT_NAME, Boolean.TRUE);
+                codegenParameter.getVendorExtensions().put(CodegenConstants.IS_LIST_CONTAINER_EXT_NAME, Boolean.TRUE);
```

Comment:
I think this statement should be removed, it does not make sense to make it there. i mean, before and after your changes.